optapy-core/pom.xml

```diff
@@ -71,6 +71,23 @@
     </plugins>
   </build>
 
+  <profiles>
+    <profile>
+      <id>stubs</id>
+      <activation>
+        <property>
+          <name>stubs</name>
+        </property>
+      </activation>
+      <dependencies>
+        <dependency>
+          <groupId>org.kie.kogito</groupId>
+          <artifactId>kogito-legacy-api</artifactId>
+        </dependency>
```

Comment:
Do we actually reference KIE API in our APIs anywhere?
(My guess is it leaked into `config`. That said, exposing foreign APIs is a bad practice. We should avoid that.)

Reply:
It in `ScoreDirectorConfig`; without it, an infinite loop occurs in stubgenj (I think the bug was recently patched) and the ScoreDirectorConfig stub is not generated.

Reply:
Looking at ScoreDirectorConfig, it does not reference KieRuntimeBuilder, yet this error occurs:
```
Skipping <java class 'org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig'> due to java.lang.NoClassDefFoundError: Lorg/kie/kogito/legacy/rules/KieRuntimeBuilder;
reference to missing class ScoreDirectorFactoryConfig - generating empty stub
```

it does not go into an infinite loop, however; the only class not generated is the ScoreDirectorFactoryConfig if we remove `kogito-legacy-api` from the stubs dependencies.